src/EventGridExtension/EventGridExtensionConfig.cs

```diff
@@ -17,22 +19,30 @@ namespace Microsoft.Azure.WebJobs.Extensions.EventGrid
     public class EventGridExtensionConfig : IExtensionConfigProvider,
                        IAsyncConverter<HttpRequestMessage, HttpResponseMessage>
     {
-        private TraceWriter _tracer = null;
+        private ILogger _logger;
+
+        internal IConverterManager ConverterManager { get; private set; }
 
         public void Initialize(ExtensionConfigContext context)
         {
             if (context == null)
             {
                 throw new ArgumentNullException("context");
             }
-            else if (context.Trace == null)
-            {
-                throw new ArgumentNullException("context.Trace");
-            }
-            _tracer = context.Trace;
+
+            _logger = context.Config.LoggerFactory.CreateLogger<EventGridExtensionConfig>();
 
             Uri url = context.GetWebhookHandler();
-            _tracer.Trace(new TraceEvent(System.Diagnostics.TraceLevel.Info, $"registered EventGrid Endpoint = {url}"));
+            _logger.LogInformation($"registered EventGrid Endpoint = {url}");
+
+            // use converterManager as a hashTable
+            // also take benefit of identity converter
+            ConverterManager = context.Config.GetService<IConverterManager>();
+            context
+                .AddConverter<JObject, string>((jobject) => jobject.ToString(Formatting.Indented))
```

Comment:
calling `AddConverter()` from context will expose my JObject converter function to everyone since I cannot specify attribute, I remember you are saying this is not a good behavior, only `ConverterManager` has methods to add Converter of concrete Attribute, and its an internal class...